source/kraken/apply_disruption.cpp

```diff
@@ -179,15 +218,23 @@ struct delete_impacts_visitor : public apply_impacts_visitor {
 
     // We set all the validity pattern to the theorical one, we will re-apply
     // other disruptions after
-    void operator()(nt::MetaVehicleJourney* mvj, const nt::Route* r = nullptr) {
-        if (impact->severity->effect == nt::disruption::Effect::NO_SERVICE) {
-            for (auto* vj: mvj->get_vjs_in_period(rt_level, impact->application_periods, meta, r)) {
-                vj->validity_patterns[rt_level] = vj->validity_patterns[nt::RTLevel::Base];
-                ++ nb_vj_reassigned;
-            }
-        } else {
-            LOG4CPLUS_DEBUG(log, "unhandled delete action on " << mvj->uri);
+    void operator()(nt::MetaVehicleJourney* mvj, nt::Route* r = nullptr) {
+        for (auto& vj: mvj->get_base_vj()) {
+            vj->validity_patterns[type::RTLevel::RealTime] =
+                    vj->validity_patterns[type::RTLevel::Adapted] =
+                            vj->validity_patterns[type::RTLevel::Base];
```

Comment:
Maybe a comment on this affectation. It looks strange and like a bug when you don't know why we do that.